jsonrpc/http.go

```diff
@@ -46,8 +47,11 @@ func (h *HTTP) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 
 	req.Body = http.MaxBytesReader(writer, req.Body, MaxRequestBodySize)
 	h.listener.OnNewRequest("any")
-	resp, err := h.rpc.HandleReader(req.Context(), req.Body)
+	resp, header, err := h.rpc.HandleReader(req.Context(), req.Body)
+
 	writer.Header().Set("Content-Type", "application/json")
+	maps.Copy(writer.Header(), header) // overwrites duplicate headers
```

Comment:
Do you think we could just do something like this instead?

writer.Header().Set(rpc.ExecutionStepsHeader, header.Get(rpc.ExecutionStepsHeader))

Reply:
we could but then we'd be heavily breaking boundaries, and not general enough.

now the http service knows about specific information concerning just a single handler.

the idea is these meta-data (headers and info returned and not part of json response but related to it - though for now just headers) can be returned by any handler.